### DISCUSSION TITLE
at global.<computed> error (multiple errors are happening to me)

### DISCUSSION THREAD

**Discussion by arashi-dev** [johnlindquist/kit#1361] [Errors]
I'm getting multiple errors these days from scriptkit.

I get the first one, when I try to run my script or the script `update-examples`:
```
at global.<computed> (file:///C:/Users/USER/.kit/target/app.js:1333:31)
exit code: 1
at ChildProcess.<anonymous> (file:///C:/Users/USER/.kit/node_modules/zx/build/core.js:146:26)
at ChildProcess.emit (node:events:513:28)
at maybeClose (node:internal/child_process:1091:16)
at ChildProcess._handle.onexit (node:internal/child_process:302:5)
at Process.callbackTrampoline (node:internal/async_hooks:130:17)
```

my script code is simple (typescript):
```ts
// Name: code
// Shortcut: control shift {

import "@johnlindquist/kit"

await $`code`
```

the 2nd error is when I try running `Ask For Help On the Forum` after an error occurs,
```
// the error points to file: C:/Users/USER/.kit/cli/error-action.js

Command failed with exit code 1: open "https://github.com/johnlindquist/kit/discussions/categories/errors"
'open' is not recognized as an internal or external command,
operable program or batch file.
Error: Command failed with exit code 1: open "https://github.com/johnlindquist/kit/discussions/categories/errors"
'open' is not recognized as an internal or external command,
operable program or batch file.
    at makeError (file:///C:/Users/USER/.kit/node_modules/@johnlindquist/globals/dist/index.js:36:111621)
    at file:///C:/Users/USER/.kit/node_modules/@johnlindquist/globals/dist/index.js:36:124733
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

the 3rd problem I have is that I cannot open the editor while choosing `edit` or `open Kenv in VS Code`. (the loader bar, loads for ever)

the other errors that I get are mostly `module X not found` while running some of the example scripts.

I tried running `Force Repair Kit SDK. Will Automatically Restart` and also reinstalled the `scriptkit` but it didn't help!

My System Information:
os: Windows 10
scriptkit version: 1.79.5
nodejs version: 18.16.0

**Comment by johnlindquist** [comment]
@arashi-dev Thanks for letting me know. I'll debug these on Windows and see what's going on.

**Reply by arashi-dev** [reply]
thank you :)

**Reply by johnlindquist** [reply]
@arashi-dev Would you mind testing the latest build to see if the errors still apply?

https://github.com/johnlindquist/kitapp/releases/tag/v1.98.9

**Reply by arashi-dev** [reply]
@johnlindquist 
No, unfortunately :(
let me check it tomorrow again (it's 5AM here and I really need to sleep). maybe I can find an error log or something.

**Reply by johnlindquist** [reply]
@arashi-dev 

The SDK log:
%USERPROFILE%\.kit\logs\kit.log

The app install/startup log:
%USERPROFILE%\AppData\Roaming\Kit\logs\main.log

**Reply by arashi-dev** [reply]
@johnlindquist 

nothing were found in those logs. however I added some console.log to to the `app.js` file as it was pointed in the first error:
```
at global.<computed> (file:///C:/Users/Arash/.kit/target/app.js:1619:31) <----- this address
exit code: 1
at ChildProcess.<anonymous> (file:///C:/Users/Arash/.kit/node_modules/zx/build/core.js:146:26)
at ChildProcess.emit (node:events:514:28)
at maybeClose (node:internal/child_process:1091:16)
at Socket.<anonymous> (node:internal/child_process:449:11)
at Socket.emit (node:events:514:28)
at Pipe.<anonymous> (node:net:323:12)
at Pipe.callbackTrampoline (node:internal/async_hooks:130:17)
```

which is this code:
```js
let loadingList = [
    "$",
    "applescript",
    "download",
    "exec",
    "fetch",
    "get",
    "patch",
    "post",
    "put",
    "del",
    "wait",
    "say",
    "playAudioFile",
];

for (let method of loadingList) {
    let original = global[method];
    global[method] = function (...args) {
        setLoading(true);
        let result = original.apply(this, args);
        result.then(() => setLoading(false));
        return result;
    };
}
```
I added a catch block to the promise and logged the error which is for the `$` function:
```json
{"_code":1,"_signal":null,"_stdout":"The system cannot find the path specified.\r\r\n","_stderr":"","_combined":"The system cannot find the path specified.\r\r\n"}
```

**Reply by johnlindquist** [reply]
@arashi-dev Try using the full path to your `code` executable. The script may not have access to the same `PATH` environment variable that you would see in your terminal (let me know if this doesn't make sense).

**Reply by arashi-dev** [reply]
@johnlindquist 
that didn't help
```ts
await $`E://arash/dev/ngrok.exe`
```
I used other exe files but nothing changed!

**Reply by johnlindquist** [reply]
@arashi-dev can you try using 'await exec()' instead "$". Want to see if it's an issue with the "$" zx package

**Reply by arashi-dev** [reply]
@johnlindquist 

Yup! It is working now:
```ts
const result = await exec(`ls`)
```

**Comment by arashi-dev** [comment]
@johnlindquist 
I've just found another error on [v1.98.9](https://github.com/johnlindquist/kitapp/releases/tag/v1.98.9)

when I try to open the script directory using `Reveal` in the script menu, it throws this error and then opens the directory:
```
applescript is Not Supported on Windows
```

should I open another discussion/issue for these types of errors I find?
